sdmetrics/single_table/privacy/dcr_overfitting_protection.py

```diff
@@ -0,0 +1,177 @@
+        num_iterations=1,
+    ):
+        """Compute the DCROverfittingProtection metric.
+
+        Args:
+            real_training_data (pd.DataFrame):
+                A pd.DataFrame object containing the real data used for training the synthesizer.
+            synthetic_data (pd.DataFrame):
+                A pandas.DataFrame object containing the synthetic data sampled
+                from the synthesizer.
+            real_validation_data (pd.DataFrame):
+                A pandas.DataFrame object containing a validation set of real data.
+                This data should not have been used to train the synthesizer.
+            metadata (dict):
+                A metadata dictionary that describes the table of data.
+            num_rows_subsample (int or None):
+                The number of synthetic data rows to subsample from the synthetic data.
+                This is used to increase the speed of the computation, if the dataset is large.
+                Defaults to None which means no subsampling will be done.
+            num_iterations (int):
+                The number of iterations to perform when subsampling.
+                The final score will be the average of all iterations. Default is 1 iteration.
+
+        Returns:
+            dict:
+                Returns a dictionary that contains the overall score, the % of synthetic data rows
+                that were closer to the validation set, and the % of synthetic data rows that were
+                closer to the real dataset. Averages of the medians are returned in the case of
+                multiple iterations.
+        """
+        sanitized_data = cls._validate_inputs(
+            real_training_data,
+            synthetic_data,
+            real_validation_data,
+            metadata,
+            num_rows_subsample,
+            num_iterations,
+        )
+
+        training_data, sanitized_synthetic_data, validation_data = sanitized_data
+
+        sum_of_scores = 0
+        sum_of_p_close_to_real = 0
+        sum_of_p_close_to_random = 0
+        for _ in range(num_iterations):
+            synthetic_sample = sanitized_synthetic_data
+            if num_rows_subsample is not None:
+                synthetic_sample = sanitized_synthetic_data.sample(n=num_rows_subsample)
+
+            dcr_real = calculate_dcr(training_data, synthetic_sample, metadata)
+            dcr_holdout = calculate_dcr(validation_data, synthetic_sample, metadata)
+
+            num_rows_closer_to_real = np.where(dcr_real < dcr_holdout, 1.0, 0.0).sum()
+            total_rows = dcr_real.size
+            percentage_close_to_real = num_rows_closer_to_real / total_rows
+            percentage_close_to_random = 1 - percentage_close_to_real
+            score = min((1.0 - percentage_close_to_real) * 2, 1.0)
+            sum_of_scores += score
+            sum_of_p_close_to_real += percentage_close_to_real
+            sum_of_p_close_to_random += percentage_close_to_random
```

Comment:
Can we be consistent with naming and rename this to `sum_percent_close_to_real`?